### PR TITLE
libevent: Build for aarch64-unknown-freebsd

### PR DESCRIPTION
**Pull request by eschnett**
I updated the OpenSSL compat bounds. Not sure whether that's necessary – it builds without that update, claiming to load `OpenSSL_jll 3.0.12`, which doesn't exist for `aarch64-unknown-freebsd`.

### CODE REVIEW

File: L/libevent/build_tarballs.jl [hunk 1]
```diff
@@ -42,7 +42,7 @@ products_unix = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("OpenSSL_jll"; compat="3.0.12"),
+    Dependency("OpenSSL_jll"; compat="3.0.15"), # Required for aarch64-unknown-freebsd
```
Comment (giordano) [inline L/libevent/build_tarballs.jl:45]: We can't change compat bounds without changing version number of the package itself.  Is there a newer version of the package?

**Review by giordano — COMMENTED** [review]
>  it builds without that update

...does this package actually need OpenSSL?

File: L/libevent/build_tarballs.jl (hunk 1 above)
Comment (eschnett) [reply L/libevent/build_tarballs.jl:45]: There is not.

**Comment by eschnett** [comment]
Yes, it needs OpenSSL. Configuring fails without OpenSSL.

**Comment by eschnett** [comment]
Configuring for Windows fails with
```
configure: error: openssl is a must but can not be found. You should add the directory containing `openssl.pc' to the `PKG_CONFIG_PATH' environment variable, or set `CFLAGS' and `LDFLAGS' directly for openssl, or use `--disable-openssl' to disable support for openssl encryption
```
However, it succeeds locally (x86_64 Linux machine running Docker).

Ideas appreciated.

**Pull request merged**